src/content/docs/en/reference/api-reference.mdx

```diff
@@ -399,6 +399,47 @@ Use [`Astro.url`](#astrourl) to construct your own canonical URL.
 
 The [canonical URL][canonical] of the current page.
 
+### `Astro.rewrite()`
+
+<p><Since v="4.13.0" /></p>
+
+Allows you to serve content from a different URL or path without redirecting the browser to a new page. 
+
+The method accepts either a string, a `URL`, or a `Request` for the location of the path.
+
+Use a string to provide an explicit path:
+
+```astro title="src/pages/index.astro"
+---
+return Astro.rewrite("/login")
+---
+```
+
+Use a `URL` type when you need to construct the URL path for the rewrite. The following example renders a page's parent path by creating a new URL from the relative  `"../"` path:
+
+```astro title="src/pages/blog/index.astro"
+---
+return Astro.rewrite(new URL("../", Astro.url))
+---
+```
+
+Use a `Request` type for complete control of the `Request` sent to the server for the new path. The following example sends a request to render the parent page while also providing headers:
+
+```astro title="src/pages/blog/index.astro"
+---
+return Astro.rewrite(new Request(new URL("../", Astro.url), {
+  headers: {
+    "x-custom-header": JSON.stringify(Astro.locals.someValue)
+  }
+}))
+---
+```
+
+The status code returned by the request will change based on the result of the rewrite. For example, if you call `Astro.rewrite("/foo/bar")`, but `/foo/bar` doesn't exist even though you have a dynamic route, Astro will return a `404` status code. 
+
+You can take advantage of this logic by conditionally return different status code. For example, for a certain user can't access certain routes, you can call `Astro.rewrite("/404")` and Astro will render the content of the 404 page (built-in or custom). 
```

Comment:
This doesn't feel very clear right now. 

Is the first paragraph anything more than routes that don't exist will 404 (like any other routes? not sure why this is special?). I don't feel like this paragraph is helpful, or provides any specific information about the rewrite. So you'll have to connect the dots a little stronger for me if this is really important!

Is the second paragraph really about conditionally *rewriting* (not about returning status codes?) If there's a 404 page, then the 404 page *itself* isn't 404'ing, right? So from the user perspective using this feature, you are rewriting with a path that does exist, just like any other path? (It just happens to be a 404 page) As written, this doesn't feel like it's about status codes since it's describing rewriting to a page that exists.

Since the other examples of usage have been very simple, if you want to talk about conditional rewrites it might be nice to actually show the code for a conditional rewrite here. Using the example of 404'ing for a restricted page is a good example, but I think it's more helpful to show what that code looks like!

(And of course, whatever we decide here, we can use for the context entry below, too, or decide just to link up here!)

Reply:
> Is the first paragraph anything more than routes that don't exist will 404 (like any other routes? not sure why this is special?). I don't feel like this paragraph is helpful, or provides any specific information about the rewrite. So you'll have to connect the dots a little stronger for me if this is really important!

Ok, I'll try to use some bridge phrase.

> Is the second paragraph really about conditionally rewriting (not about returning status codes?)

It is **not** about conditional rewriting, it's about status code and expectations, but it's a bit silly to say if you do this, this other thing happen. So, I was trying to put some real usage context. Let me see if I can do something better with an example.

Reply:
@sarah11918 please have a look at this new version [`cc081b4` (#8914)](https://github.com/withastro/docs/pull/8914/commits/cc081b450ce5260672b9b3f215dba4df9727c460)